tor/cmd_onion.go

```diff
@@ -254,3 +257,38 @@ func (c *Controller) AddOnion(cfg AddOnionConfig) (*OnionAddr, error) {
 		Port:         cfg.VirtualPort,
 	}, nil
 }
+
+// DelOnion tells the server to remove an Onion ("Hidden") Service, that was
+// previously created via an "ADD_ONION" command. It is only possible to remove
+// Onion Services that were created on the same control connection as the
+// "DEL_ONION" command, and those that belong to no control connection in
+// particular (The "Detach" flag was specified at creation).
+func (c *Controller) DelOnion(serviceID string) error {
+	log.Debugf("removing serviceID:%s from tor controller", serviceID)
+
+	cmd := fmt.Sprintf("DEL_ONION %s", serviceID)
+
+	// Send the command to create the onion service to the Tor server and
+	// await its response.
+	code, _, err := c.sendCommand(cmd)
+
+	// Tor replies with "250 OK" on success, or a 512 if there are an
```

Comment:
Say you're running lnd and then upgrade to this change, is it possible that we try to `DelOnion` on an onion service that does not belong to a control connection and doesn't have "Detatch" set? (ie, was created before this PR)

Wondering what error code that returns/ whether we need to handle it?

Reply:
Yeah it was a bit confusing, I've updated the docs on `AddOnion`, hopefully that would make things a bit clearer.

For the specific case, no it won't happen as,
- when running old lnd, the old Onion Service would belong to the old control connection, so it cannot be accessed by other connections.
- when upgrading, lnd needs to be restarted, thus the control connection will be re-created, and the old Onion Service would be gone, a new one will be created.

From the specs, there are three possible response codes we can get, 250 on success, 512 on invalid arguments, and 552 if the `ServiceID` is not recognized. I've updated the docs below to make it more verbose on code 552.